### ISSUE TITLE
feature: emulate usb device

### ISSUE THREAD

**Issue by antonilol** [nspire-emus/firebird#322]
emulate a usb device so other programs can interact with it like a hardware calculator

**Comment by Vogtinator** [comment]
On Linux this is theoretically possible with a "vhci" controller, but on others I'm not aware of any API that could be used.

**Comment by gfhdhytghd** [comment]
yeah it need a driver to emulate usb on windows, but I wish I could see this function on linux soon

**Comment by Vogtinator** [comment]
There is https://github.com/cezanne/usbip-win, maybe that works well enough. If firebird exposes a USB/IP server, this could be used from Linux and Windows.

**Comment by gfhdhytghd** [comment]
yeah usb-over-ip could be a good solution, it will be easier to cross-platform development, and it can even use on android and macos
